### ISSUE TITLE
cannot build

### ISSUE THREAD

**Issue by chenmoucheng** [martinra/hlinear#3]
Hi,

When I tried `stack build`, I got the following error message:

```
[49 of 68] Compiling HLinear.NormalForm.FoldUnfold.ReduceEchelonForm.FractionFree ( src/HLinear/NormalForm/FoldUnfold/ReduceEchelonForm/FractionFree.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/HLinear/NormalForm/FoldUnfold/ReduceEchelonForm/FractionFree.o )

src/HLinear/NormalForm/FoldUnfold/ReduceEchelonForm/FractionFree.hs:50:19: error:
    • No instance for (MonadZip Vector)
        arising from a use of ‘mzipWith’
    • In the expression:
        mzipWith (\ a an -> mulSubMul a den an r2) r3 rn3
      In an equation for ‘r3'’:
          r3' = mzipWith (\ a an -> mulSubMul a den an r2) r3 rn3
      In an equation for ‘normalize’:
          normalize (EchelonFormRow on rn) (EchelonFormRow o r, s)
            = (EchelonFormRow o $ r1 <> zero `V.cons` r3' <> r4', on)
            where
                (r1, r234) = V.splitAt (on - o) r
                r34 = V.tail r234
                (r23, r4) = V.splitAt (s - on) r234
                r2 = V.head r23
                ....
```

Could you let me know how to fix it?  Thanks for any help you can provide!

**Comment by martinra** [comment]
Thanks for your feedback. It builds OK for me with lts-9.0 and extra-deps: permutation-0.5.0.5. Which version do you use?

**Comment by chenmoucheng** [comment]
Thanks for the prompt response.  I guess I am using lts-8.24 without permutation:

```
Looking for .cabal or package.yaml files to use to init the project.
Using cabal packages:
- algebraic-structures/
- hflint/
- hlinear/
- natural-test/
- vector-test/


Warning: Cabal file warning in /Users/doug/Desktop/hlinear/hflint/hflint.cabal@ 5:9: Tabs used as indentation at 5:9, 6:8
Selecting the best among 13 snapshots...

* Rejected lts-11.7
    ghc-8.2.2 cannot be used for these packages:
        - natural-test
        - vector-test
    base version 4.10.1.0 found
        - natural-test requires >=4.8 && <4.10
        - vector-test requires >=4.8 && <4.10

* Rejected nightly-2018-04-30
    ghc-8.4.2 cannot be used for these packages:
        - natural-test
        - vector-test
    base version 4.11.1.0 found
        - natural-test requires >=4.8 && <4.10
        - vector-test requires >=4.8 && <4.10

* Rejected lts-10.10
    ghc-8.2.2 cannot be used for these packages:
        - natural-test
        - vector-test
    base version 4.10.1.0 found
        - natural-test requires >=4.8 && <4.10
        - vector-test requires >=4.8 && <4.10

* Partially matches lts-9.21
    permutation not found
        - hlinear requires -any

* Matches lts-8.24

Selected resolver: lts-8.24
Initialising configuration using resolver: lts-8.24
Total number of user packages considered: 5
Overwriting existing configuration file: stack.yaml
All done.
```

**Comment by martinra** [comment]
master now fixes LTS 9.21 as build environment and builds without modifications from scratch.
